File: libs/eave-stdlib-ts/src/github-api/operations/query-repos.ts
import { RequestArgsTeamId, makeRequest } from "../../requests.js";
import { ExternalGithubRepo } from "../models.js";
import { GithubAppEndpointConfiguration } from "./shared.js";

export type QueryGithubReposResponseBody = {
  repos: ExternalGithubRepo[];
};

export class QueryGithubReposOperation {
  static config = new GithubAppEndpointConfiguration({
    path: "/github/api/repos/query",
    authRequired: false,
  });

  static async perform(
    args: RequestArgsTeamId,
  ): Promise<QueryGithubReposResponseBody> {
    const resp = await makeRequest({
      url: this.config.url,
      ...args,
    });
    const responseData = <QueryGithubReposResponseBody>await resp.json();
    return responseData;
  }
}

File: libs/eave-stdlib-py/src/eave/stdlib/github_api/operations/query_repos.py
import uuid
from typing import Unpack

from eave.stdlib import requests
from eave.stdlib.endpoints import BaseResponseBody
from eave.stdlib.github_api.models import ExternalGithubRepo
from eave.stdlib.github_api.operations import GithubAppEndpoint, GithubAppEndpointConfiguration


class QueryGithubRepos(GithubAppEndpoint):
    config = GithubAppEndpointConfiguration(
        path="/github/api/repos/query",
    )

    class ResponseBody(BaseResponseBody):
        repos: list[ExternalGithubRepo]

    @classmethod
    async def perform(cls, team_id: uuid.UUID | str, **kwargs: Unpack[requests.CommonRequestArgs]) -> ResponseBody:
        response = await requests.make_request(
            config=cls.config,
            input=None,
            team_id=team_id,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body
